### PR TITLE
Feat/add user input code context

### PR DESCRIPTION
**Pull request by welingtonsampaio**
# Enhance Context Awareness for Commit Message Generation

OpenCommit now allows you to provide additional context when generating commit messages. This context is incorporated into the prompt given to the AI model, enabling it to create more tailored and relevant commit messages.

## Usage

To provide additional context, use the `--context` flag followed by your context in quotes after the OpenCommit command:

```
oco --context "This commit updates the authentication flow to use JWT tokens"
```

The AI model will consider the provided context when generating the commit message, incorporating relevant information when appropriate.

## Benefits

- **Context-aware commit messages**: By providing additional context, the generated commit messages will be more aligned with the specific changes made, resulting in clearer and more descriptive commit messages.
- **Improved relevance**: The AI model can better understand the intent and purpose behind the changes, leading to commit messages that accurately reflect the modifications.
- **Customization**: You can tailor the commit message generation process by providing context specific to your project or development environment.

This enhancement strengthens OpenCommit's ability to generate meaningful and accurate commit messages, facilitating better commit history management and code collaboration.

### CODE REVIEW

**Comment by di-sukharev** [comment]
thats awesome, reading

File: src/commands/commit.ts [hunk 1]
```diff
@@ -36,6 +36,16 @@ const checkMessageTemplate = (extraArgs: string[]): string | false => {
   return false;
 };
 
+// remove all args after '--'
```
Comment (di-sukharev) [inline src/commands/commit.ts:39]: pls remove comments

Comment (di-sukharev) [reply src/commands/commit.ts:39]: or make them answer `why` if needed

File: src/prompts.ts [hunk 2]
```diff
@@ -111,6 +111,26 @@ const getOneLineCommitInstruction = () =>
...
 
+/**
+ * Get the context of the user input
+ * @param extraArgs - The arguments passed to the command line
+ * @example
+ *  $ oco -- This is a context used to generate the commit message
+ * @returns - The context of the user input
+ */
+const userInputCodeContext = () => {
+  const args = process.argv;
```
Comment (di-sukharev) [inline src/prompts.ts:122]: i think this function should accept an argument `context: string`, wdyt?

File: src/prompts.ts [hunk 3]
```diff
@@ -111,6 +111,26 @@ const getOneLineCommitInstruction = () =>
...
+  // Find all arguments after '--'
+  const dashIndex = args.indexOf('--');
+  if (dashIndex !== -1) {
+    const context = args.slice(dashIndex + 1).join(' ');
+    if (context !== '' && context !== ' ') {
+      return `Additional context provided by the user: ${context}\n\nConsider this context when generating the commit message, incorporating relevant information when appropriate.`;
```
Comment (di-sukharev) [inline src/prompts.ts:128]: 🫡

**Review by di-sukharev — COMMENTED** [review]
great, left few comments

File: src/prompts.ts (hunk 2 above)
Comment (welingtonsampaio) [reply src/prompts.ts]: To implement this in a more effective manner, I switched to a flag called context. This approach makes the code more declarative and self-explanatory. Users can learn about this feature by running the oco --help command, which provides detailed information on how to use the context flag and its purpose within the application

Comment (welingtonsampaio) [reply src/prompts.ts]: Check again when you can

File: src/commands/commit.ts [hunk 4]
```diff
@@ -197,6 +199,7 @@ ${chalk.grey('——————————————————')}`
 
 export async function commit(
   extraArgs: string[] = [],
+  context: string = '',
```
Comment (di-sukharev) [inline src/commands/commit.ts:202]: exactly

**Comment by di-sukharev** [comment]
nice job, i will be merging tmr

**Pull request merged**